api/src/functions/root/api-upload-csv/index.ts

```diff
@@ -0,0 +1,324 @@
+
+interface IErrorLog {
+    row: number,
+    errorMessages: Array<string>
+}
+
+const numberOfTopAffectedGroups:number = 3;
+const numberOfTopPrioritySectors:number = 3;
+const numberOfTopVulnerableGroups:number = 3;
+const numberOfTopResponseModalities:number = 3;
+
+
+
+module.exports = function (context:any,req:any):void {
+    context.log('JavaScript HTTP trigger function processed a request.');
+    
+    let rowNumber:number = 0;
+    let errorLog:IErrorLog[] = [];
+    
+    const requestBody = req.body;
+    fast_csv.fromString(requestBody, {
+        objectMode:true,
+        headers:true,
+        trim:true
+    }).validate(function (data:IUserInfo):boolean {
+        let validationErrorLog:Array<string> = [];
+        rowNumber++;
+
+        validateHeaders(data,validationErrorLog);
+        validateUserInfo(data,validationErrorLog);
+        validateAffectedPeople(data,validationErrorLog);
+        validateSectorSeverity(data,validationErrorLog);
+        validateSectorFactorData(data,validationErrorLog);
+        validateSectorBasicNeedsConcern(data,validationErrorLog);
+        validateTopAffectedGroups(data,validationErrorLog);
+        validateTopPrioritySectors(data,validationErrorLog);
+        validateTopVulnerableGroups(data,validationErrorLog);
+        validateTopResponseModalities(data,validationErrorLog);
+        if (validationErrorLog.length === 0) {
+            return true;
+        } else {
+            errorLog.push(
+                {
+                    row: rowNumber,
+                    errorMessages: validationErrorLog
+                }
+            );
+            return false;
+        }
+    }).on("data-invalid", function (data:any):void {
+        context.log("Data is Invalid! Please review the errors below.");
+    }).on("data",function (data:any):void {
+        
+        // TODO Save the report in cosmos
+        
+        context.log(JSON.stringify(createReport(data)));
+    }).on("end",function ():void {
+        let csvUploadResponse:object = {};
+        if (errorLog.length === 0) {
+            context.log("Data has been imported successfully!");
+            csvUploadResponse["Message"] = "Data has been imported successfully!";
+            context.res = {status: 200, body: csvUploadResponse};
+        } else {
+            context.log("There were various errors. Please review them below - ");
+            csvUploadResponse["Message"] = "There were various errors. Please review them below - ";
+            csvUploadResponse["Errors"] = errorLog;
+            context.res = {status: 422, body: csvUploadResponse};
+        }
+        context.done();
+    })
+}
+
+function validateHeaders(data:object,errorLog:Array<string>):void {
+    const csvProperties:Array<string> = Object.getOwnPropertyNames(data);
+
+    for (const header of csvProperties) {
+        if (!csvHeaders.includes(header)) {
+            errorLog.push(`Following CSV Header was not recognized [${header}]`);
```

Comment:
Let's change this slightly to `CSV header was not recognized: '${header}'`

Reply:
Done.